Content.Shared/SSDIndicator/SSDIndicatorSystem.cs

```diff
@@ -0,0 +1,27 @@
+﻿using Content.Shared.Mind.Components;
+
+namespace Content.Shared.SSDIndicator;
+
+/// <summary>
+///     Handle changing player SSD indicator status
+/// </summary>
+public sealed class SSDIndicatorSystem : EntitySystem
+{
+    public override void Initialize()
+    {
+        SubscribeLocalEvent<SSDIndicatorComponent, MindAddedMessage>(OnMindAdded);
+        SubscribeLocalEvent<SSDIndicatorComponent, MindRemovedMessage>(OnMindRemoved);
+    }
+
+    private void OnMindAdded(EntityUid uid, SSDIndicatorComponent component, MindAddedMessage args)
+    {
+        component.IsSSD = false;
+        Dirty(uid, component);
+    }
+
+    private void OnMindRemoved(EntityUid uid, SSDIndicatorComponent component, MindRemovedMessage args)
+    {
+        component.IsSSD = true;
+        Dirty(uid, component);
+    }
```

Comment:
This should also check if the entity has an NPC component as other mobs should probably get it.